analyzers/src/SonarAnalyzer.CSharp/Rules/DeadStores.cs

```diff
@@ -330,22 +285,22 @@ private void ProcessPostfixExpression(SyntaxNode instruction, HashSet<ISymbol> l
                 if (operand.IsKind(SyntaxKind.IdentifierName)
                     && context.SemanticModel.GetSymbolInfo(operand).Symbol is { } symbol
                     && IsSymbolRelevant(symbol)
-                    && CSharpLiveVariableAnalysis.IsLocalScoped(symbol, declaration)
+                    && CSharpLiveVariableAnalysis.IsLocalScoped(symbol, nodeSymbol)
                     && !liveOut.Contains(symbol)
                     && !IsMuted(operand))
                 {
-                    context.ReportDiagnosticWhenActive(Diagnostic.Create(rule, postfixExpression.GetLocation(), symbol.Name));
+                    context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, postfixExpression.GetLocation(), symbol.Name));
                 }
             }
 
             private void ReportOnAssignment(AssignmentExpressionSyntax assignment, ExpressionSyntax left, ISymbol symbol, HashSet<SyntaxNode> assignmentLhs, HashSet<ISymbol> outState)
             {
-                if (CSharpLiveVariableAnalysis.IsLocalScoped(symbol, declaration)
+                if (CSharpLiveVariableAnalysis.IsLocalScoped(symbol, this.nodeSymbol)
```

Comment:
Nitpick: this is redundant.
```suggestion
                if (CSharpLiveVariableAnalysis.IsLocalScoped(symbol, nodeSymbol)
```